interbtc-spec/docs/source/spec/nomination.rst

```diff
@@ -208,463 +131,154 @@ Specification
-
-* ``WithdrawalRequestNotFound``: no withdrawal request found for the given id.
-
-Preconditions
-.............
-
-* The BTC Parachain status in the :ref:`security` component must be set to ``RUNNING:0``.
-
-
-Function Sequence
-.................
-
-1. Check ``operatorId`` is an operator. If not, throw ``VaultNotOptedInToNomination``.
-
-2. Check ``requestId`` corresponds to an actual withdrawal request. If not, throw ``WithdrawalRequestNotFound``.
-
-3. Remove the withdrawal request from the ``withdrawalRequests`` array in the ``Operator`` struct for ``operatorId``.
-
-4. Increase the backing collateral of ``operatorId`` in the :ref:`Vault-registry` by the amount in the withdrawal request.
-
-
-.. _cancelNominatorWithdrawal:
-
-cancelNominatorWithdrawal
--------------------------
+withdrawCollateral
+------------------
 
-Cancel a nominator's withdrawal request.
+Withdraw collateral from a nominated Vault.
 
 Specification
 .............
 
 *Function Signature*
 
-``cancelNominatorWithdrawal(nominatorId, operatorId, requestId)``
+``withdrawCollateral(vaultId, nominatorId, amount)``
 
 *Parameters*
 
-* ``nominatorId``: the id of the nominator.
-
-* ``operatorId``: the id of the operator.
-
-* ``requestId``: the id of the withdrawal request.
+* ``vaultId``: the id of the previously nominated vault.
+* ``nominatorId``: the id of the user who nominated collateral.
+* ``amount``: the amount of collateral to withdraw.
 
 *Events*
 
-* ``CancelNominatorCollateralWithdrawal(requestId, nominatorId, operatorId)``
+* ``WithdrawCollateral(vaultId, nominatorId, amount)``
 
-*Errors*
+*Preconditions*
 
-* ``VaultNotOptedInToNomination``: the ``operatorId`` is not an Operator.
+* The BTC Parachain status in the :ref:`security` component MUST NOT be ``SHUTDOWN:2``.
+* A vault with id ``vaultId`` MUST be registered.
+* A vault with id ``vaultId`` MUST have nomination enabled.
 
-* ``NominatorNotFound``: the ``nominatorId`` is not a Nominator.
+*Postconditions*
 
-* ``WithdrawalRequestNotFound``: no withdrawal request found for the given id.
-
-Preconditions
-.............
-
-* The BTC Parachain status in the :ref:`security` component must be set to ``RUNNING:0``.
-
-
-Function Sequence
-.................
-
-1. Check ``operatorId`` is an operator. If not, throw ``VaultNotOptedInToNomination``.
-
-2. Check ``nominatorId`` is a nominator. If not, throw ``NominatorNotFound``.
-
-3. Check ``requestId`` corresponds to an actual withdrawal request. If not, throw ``WithdrawalRequestNotFound``.
-
-3. Remove the withdrawal request from the ``withdrawalRequests`` array in the ``Nominator`` struct inside the ``Operator`` struct for ``operatorId``.
-
-4. Increase the backing collateral of ``operatorId`` in the :ref:`Vault-registry` by the amount in the withdrawal request.
+* The vault's collateral MUST decrease by the amount withdrawn.
+* The nominators MUST earn rewards equivalent to their collateral.
```

Comment:
It seems like the vault withdraws rewards, not the nominator? This is the call being made: `withdraw_all_vault_rewards::<T>(&vault_id)`

I don't really understand how this works..

Reply:
Missing postcondition: stake change